simplify image content and bundles

pohly opened the pull request:
Removes the extra "ostro-image-swupd-reference" image and the
"reference" bundle and instead moves the content into the base images
of both ostro-image-swupd and ostro-image-noswupd.

Because the doc changes would conflict with the changes in PR #100,
this PR here includes all of those commits, too. That way we can
merge PR #100 and then this one, or this one directly.

@barbieri I hope this addresses your review feedback from PR #95.

@olev @mikko @kad I expect this to speed up the builds a bit.

barbieri commented on doc/howtos/building-images.rst:
```diff
@@ -78,6 +81,51 @@ the existing user accounts (including root) has a password set, so
+-----------------
+
+When building production images, first follow the instructions
+provided in :file:`meta-intel-iot-security/meta-integrity/README.md` for creating your own
```
@dbkinder and @gvancuts should we convert that doc to `.rst` and then integrate it in our website autogen documentation?

dbkinder replied:
Yes, this should be included in the documentation set.

README files are OK for some explanatory information about the component, but we should not have users wandering around components looking for them.  We (Geoffroy/David) should take a look around the other components and see if there are substantial gems of information scattered around in README files we can incorporate into the documentation set.


-- david

From: Gustavo Sverzut Barbieri [mailto:notifications@github.com]
Sent: Tuesday, April 12, 2016 7:27 AM
To: ostroproject/meta-ostro <meta-ostro@noreply.github.com>
Cc: Kinder, David B <david.b.kinder@intel.com>
Subject: Re: [ostroproject/meta-ostro] simplify image content and bundles (#104)


In doc/howtos/building-images.rst<https://github.com/ostroproject/meta-ostro/pull/104#discussion_r59385123>:

> +Developers building their own images for personal use can follow these

> +instructions to replicate the configuration of the published Ostro OS images. All necessary

> +private keys are provided in the ``ostro-os`` repository.

> +

> +To do this, before building,  edit the :file:`conf/local.conf` configuration file,

> +find the line

> +with ``# require conf/distro/include/ostro-os-development.inc`` and

> +uncomment it. This will also add some recommended software to the ``ostro-image-noswupd``

> +reference image, see below for details.

> +

> +

> +Production Images

> +-----------------

> +

> +When building production images, first follow the instructions

> +provided in :file:`meta-intel-iot-security/meta-integrity/README.md` for creating your own

@dbkinder<https://github.com/dbkinder> and @gvancuts<https://github.com/gvancuts> should we convert that doc to .rst and then integrate it in our website autogen documentation?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/ostroproject/meta-ostro/pull/104/files/706737ae1b53d4576a0c68c3f4ebcb961063b620#r59385123>

gvancuts replied:
+1 on adding a subset of the README.md file to our main doc (via .rst file).

barbieri commented on doc/howtos/building-images.rst:
```diff
@@ -171,24 +194,66 @@ Example::
+by using image recipes like ``ostro-image-noswupd`` where swupd is
+turned off, because:
+
+a) swupd support is still new and may have unexpected problems.
```
I'd not write this in the documentation :-D

pohly replied:
But it's true, so why not call the child by its name? ;-} Anyone claiming that swupd support will be rock-solid right after integrating it is, well, very optimistic.

If anyone has suggestions for a better working of that point, I'm all ears.

I also don't mind taking out the entire point a).

barbieri replied:
I'd take it out... what I see in practice is that thing "sticks" and then it's hard to remove the bad karma :-)

All parts of the system may have bugs, in that case they will be reported and fixed.

pohly replied:
Resolved by taking it out in PR #100 in commit da1b663c60.

barbieri commented:
+1 for me, very good.

pohly commented:
PR #100 is currently more recent than this PR here. I suggest we merge PR #100 first, then rebase this one here and merge it later.

pohly commented:
Rebased on top of the recent app-framework image feature change and also fixed that (must be added to OSTRO_IMAGE_PKG_FEATURES). It also turned out that "full" is a reserved bundle name (with some subtle implications, like include os-core in pack-full-from-0.tar when it should have contained only the additional files), so now ostro-image-swupd.bb uses "world" as bundle name for "all our additional software".
